### ISSUE TITLE
CVE-2022-1996

### ISSUE THREAD

**Issue by m-barthelemy** [TremoloSecurity/kube-oidc-proxy#20]
Our security scans report that kube-oidc-proxy 1.0.1 is impacted by CVE-2022-1996 (https://nvd.nist.gov/vuln/detail/CVE-2022-1996) for using the Go-restful library.

Would the project consider updating this dependency?

**Comment by mlbiam** [comment]
I think this does the trick.  Snyk is saying anything below 3.8 is impacted, but github says on the 2.x branch 2.9.5 is OK.  Can you scan https://github.com/mlbiam/kube-oidc-proxy to see if it resolves the issue?  If so we'll get a release cut

**Comment by m-barthelemy** [comment]
Thanks for the update!
Unfortunately Trivy is still reporting the same vulnerability - not sure if it's because it's real, or if it's a bug in Trivy itself.

```bash

matthieu.barthelemy@matthieu git % git clone git@github.com:mlbiam/kube-oidc-proxy.git
Cloning into 'kube-oidc-proxy'...
remote: Enumerating objects: 63859, done.
remote: Counting objects: 100% (644/644), done.
remote: Compressing objects: 100% (270/270), done.
remote: Total 63859 (delta 327), reused 596 (delta 314), pack-reused 63215
Receiving objects: 100% (63859/63859), 105.46 MiB | 1.04 MiB/s, done.
Resolving deltas: 100% (26717/26717), done.
matthieu.barthelemy@matthieu git % cd kube-oidc-proxy 
matthieu.barthelemy@matthieu kube-oidc-proxy % ~/Downloads/trivy_0.29.1_macOS-ARM64/trivy fs .
2022-07-25T08:53:32.904+0400	INFO	Need to update DB
2022-07-25T08:53:32.904+0400	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2022-07-25T08:53:32.904+0400	INFO	Downloading DB...
33.20 MiB / 33.20 MiB [---------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 2.44 MiB p/s 14s
2022-07-25T08:53:51.067+0400	INFO	Vulnerability scanning is enabled
2022-07-25T08:53:51.067+0400	INFO	Secret scanning is enabled
2022-07-25T08:53:51.067+0400	INFO	If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2022-07-25T08:53:51.067+0400	INFO	Please see also https://aquasecurity.github.io/trivy/v0.29.1/docs/secret/scanning/#recommendation for faster secret detection
2022-07-25T08:53:52.138+0400	INFO	Number of language-specific files: 1
2022-07-25T08:53:52.138+0400	INFO	Detecting gomod vulnerabilities...

go.mod (gomod)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 1)

┌────────────────────────────────┬────────────────┬──────────┬───────────────────────────────────┬───────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│            Library             │ Vulnerability  │ Severity │         Installed Version         │           Fixed Version           │                            Title                             │
├────────────────────────────────┼────────────────┼──────────┼───────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ github.com/emicklei/go-restful │ CVE-2022-1996  │ CRITICAL │ 2.9.5+incompatible                │ v3.8.0                            │ go-restful: Authorization Bypass Through User-Controlled Key │
│                                │                │          │                                   │                                   │ https://avd.aquasec.com/nvd/cve-2022-1996                    │
├────────────────────────────────┼────────────────┼──────────┼───────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/crypto            │ CVE-2022-27191 │ HIGH     │ 0.0.0-20220214200702-86341886e292 │ 0.0.0-20220314234659-1baeb1ce4c0b │ golang: crash in a golang.org/x/crypto/ssh server            │
│                                │                │          │                                   │                                   │ https://avd.aquasec.com/nvd/cve-2022-27191                   │
└────────────────────────────────┴────────────────┴──────────┴───────────────────────────────────┴───────────────────────────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by mlbiam** [comment]
@m-barthelemy thanks for the scan results.  i think we have this figured out.  no longer seeing either of these in trivy or in snyk.  local tests passed, need to run through our QA process before release.

**Comment by mlbiam** [comment]
published 1.0.2 which fixes this issue
